src/xdocs/config_coding.xml

```diff
@@ -1961,6 +1985,23 @@ class PageBuilder {
   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <source>
+public class SomeClass {
+
+  private String field;
+
+  public void method(String param) {
+    String field = param; // violation, 'field' variable hides 'field' field
+  }
+
+  public void method2(String field) // OK
+  {
+    String var = field;
+  }
+
+  public static void main(String args[]) { }
+}
+        </source>
```

Comment:
this code is not compiling without main.

Reply:
Code can compile without main method. Something is wrong with way you check it. Even code like `class A {}` will compile

Reply:
made some changes please check now